### PR TITLE
remove unnecessary community client from resource test files

### PR DESCRIPTION
**Pull request by fanny-jiang**
This PR removes unnecessary use of the community client from the Monitors and Metric Metadata resource test files and updates them to use the generated `datadog-api-client-go`. It also includes some minor cleanup of unused parameters. 

Note: the `checkPostEvent` function still relies the community client as the [`datadog-api-client-go`](https://github.com/DataDog/datadog-api-client-go/blob/master/api/v1/datadog/docs/MetricsApi.md#metricsapi) doesn't currently support the `POST` endpoint.

### CODE REVIEW

File: datadog/resource_datadog_metric_metadata_test.go [hunk 1]
```diff
@@ -94,24 +96,36 @@ func TestAccDatadogMetricMetadata_Updated(t *testing.T) {
 	})
 }
 
+func metadataExistsHelper(s *terraform.State, datadogV1Client *datadogV1.APIClient, authV1 context.Context) error {
```
Comment (skarimo) [inline datadog/resource_datadog_metric_metadata_test.go:99]:
```suggestion
func metadataExistsHelper(s *terraform.State, datadogClientV1 *datadogV1.APIClient, authV1 context.Context) error {
```
To maintain same format as rest of the repo

File: datadog/resource_datadog_metric_metadata_test.go [hunk 2]
```diff
@@ -94,24 +96,36 @@ func TestAccDatadogMetricMetadata_Updated(t *testing.T) {
...
+	for _, r := range s.RootModule().Resources {
+		metric, ok := r.Primary.Attributes["metric"]
+		if !ok {
+			continue
+		}
+
+		_, _, err := datadogV1Client.MetricsApi.GetMetricMetadata(authV1, metric).Execute()
+
```
Comment (skarimo) [inline datadog/resource_datadog_metric_metadata_test.go:107]:
```suggestion
```

File: datadog/resource_datadog_metric_metadata_test.go [hunk 3]
```diff
@@ -94,24 +96,36 @@ func TestAccDatadogMetricMetadata_Updated(t *testing.T) {
...
-		client := providerConf.CommunityClient
-		for _, r := range s.RootModule().Resources {
-			metric, ok := r.Primary.Attributes["metric"]
-			if !ok {
-				continue
-			}
-			if _, err := client.ViewMetricMetadata(metric); err != nil {
-				return fmt.Errorf("received an error retrieving metric_metadata %s", err)
-			}
+		datadogV1Client := providerConf.DatadogClientV1
```
Comment (skarimo) [inline datadog/resource_datadog_metric_metadata_test.go:118]:
```suggestion
		datadogClientV1 := providerConf.DatadogClientV1
```

File: datadog/resource_datadog_metric_metadata_test.go [hunk 4]
```diff
@@ -94,24 +96,36 @@ func TestAccDatadogMetricMetadata_Updated(t *testing.T) {
...
+		authV1 := providerConf.AuthV1
+
+		if err := metadataExistsHelper(s, datadogV1Client, authV1); err != nil {
```
Comment (skarimo) [inline datadog/resource_datadog_metric_metadata_test.go:121]:
```suggestion
		if err := metadataExistsHelper(s, datadogClientV1, authV1); err != nil {
```
Here as well

File: datadog/resource_datadog_monitor_test.go [hunk 5]
```diff
@@ -683,20 +683,23 @@ func TestAccDatadogMonitor_ComposeWithSyntheticsTest(t *testing.T) {
 func testAccCheckDatadogMonitorDestroy(accProvider *schema.Provider) func(*terraform.State) error {
 	return func(s *terraform.State) error {
 		providerConf := accProvider.Meta().(*ProviderConfiguration)
-		client := providerConf.CommunityClient
+		datadogV1Client := providerConf.DatadogClientV1
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:686]:
```suggestion
		datadogClientV1 := providerConf.DatadogClientV1
```

File: datadog/resource_datadog_monitor_test.go [hunk 6]
```diff
@@ -683,20 +683,23 @@ func TestAccDatadogMonitor_ComposeWithSyntheticsTest(t *testing.T) {
...
+		authV1 := providerConf.AuthV1
 
-		if err := destroyHelper(s, client); err != nil {
+		if err := destroyHelper(s, datadogV1Client, authV1); err != nil {
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:689]:
```suggestion
		if err := destroyHelper(s, datadogClientV1, authV1); err != nil {
```

File: datadog/resource_datadog_monitor_test.go [hunk 7]
```diff
@@ -683,20 +683,23 @@ func TestAccDatadogMonitor_ComposeWithSyntheticsTest(t *testing.T) {
...
 	}
 }
 
-func testAccCheckDatadogMonitorExists(accProvider *schema.Provider, n string) resource.TestCheckFunc {
+func testAccCheckDatadogMonitorExists(accProvider *schema.Provider) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		providerConf := accProvider.Meta().(*ProviderConfiguration)
-		client := providerConf.CommunityClient
-		if err := existsHelper(s, client); err != nil {
+		datadogV1Client := providerConf.DatadogClientV1
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:699]:
```suggestion
		datadogClientV1 := providerConf.DatadogClientV1
```

File: datadog/resource_datadog_monitor_test.go [hunk 8]
```diff
@@ -683,20 +683,23 @@ func TestAccDatadogMonitor_ComposeWithSyntheticsTest(t *testing.T) {
...
+		authV1 := providerConf.AuthV1
+
+		if err := existsHelper(s, datadogV1Client, authV1); err != nil {
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:702]:
```suggestion
		if err := existsHelper(s, datadogClientV1, authV1); err != nil {
```

File: datadog/resource_datadog_monitor_test.go [hunk 9]
```diff
@@ -1254,24 +1257,29 @@ resource "datadog_monitor" "foo" {
...
 			return fmt.Errorf("received an error retrieving monitor %s", err)
 		}
 		return fmt.Errorf("monitor still exists")
 	}
+
 	return nil
 }
 
-func existsHelper(s *terraform.State, client *datadog.Client) error {
+func existsHelper(s *terraform.State, datadogV1Client *datadogV1.APIClient, authV1 context.Context) error {
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:1278]:
```suggestion
func existsHelper(s *terraform.State, datadogClientV1 *datadogV1.APIClient, authV1 context.Context) error {
```

File: datadog/resource_datadog_monitor_test.go [hunk 10]
```diff
@@ -1254,24 +1257,29 @@ resource "datadog_monitor" "foo" {
...
 	for _, r := range s.RootModule().Resources {
-		i, _ := strconv.Atoi(r.Primary.ID)
-		if _, err := client.GetMonitor(i); err != nil {
+		i, _ := strconv.ParseInt(r.Primary.ID, 10, 64)
+		_, _, err := datadogV1Client.MonitorsApi.GetMonitor(authV1, i).Execute()
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:1281]:
```suggestion
		_, _, err := datadogClientV1.MonitorsApi.GetMonitor(authV1, i).Execute()
```

File: datadog/resource_datadog_monitor_test.go [hunk 11]
```diff
@@ -1254,24 +1257,29 @@ resource "datadog_monitor" "foo" {
...
-func destroyHelper(s *terraform.State, client *datadog.Client) error {
+func destroyHelper(s *terraform.State, datadogClientV1 *datadogV1.APIClient, authV1 context.Context) error {
+
 	for _, r := range s.RootModule().Resources {
-		i, _ := strconv.Atoi(r.Primary.ID)
-		if _, err := client.GetMonitor(i); err != nil {
-			if strings.Contains(err.Error(), "404 Not Found") {
+		i, _ := strconv.ParseInt(r.Primary.ID, 10, 64)
+		_, httpresp, err := datadogClientV1.MonitorsApi.GetMonitor(authV1, i).Execute()
+
```
Comment (skarimo) [inline datadog/resource_datadog_monitor_test.go:1265]:
```suggestion
```

**Review by skarimo — COMMENTED** [review]
Small nits

File: datadog/resource_datadog_metric_metadata_test.go [hunk 12]
```diff
@@ -94,24 +96,35 @@ func TestAccDatadogMetricMetadata_Updated(t *testing.T) {
...
+func metadataExistsHelper(s *terraform.State, datadogClientV1 *datadogV1.APIClient, authV1 context.Context) error {
+	for _, r := range s.RootModule().Resources {
+		metric, ok := r.Primary.Attributes["metric"]
+		if !ok {
+			continue
+		}
+
+		_, _, err := datadogV1Client.MetricsApi.GetMetricMetadata(authV1, metric).Execute()
+		if err != nil {
+			return fmt.Errorf("received an error retrieving metric_metadata %s", err)
```
Comment (zippolyte) [inline datadog/resource_datadog_metric_metadata_test.go:108]:
```suggestion
			return translateClientError(err, "error retrieving metric_metadata")
```

File: datadog/resource_datadog_monitor_test.go [hunk 13]
```diff
@@ -1254,24 +1257,28 @@ resource "datadog_monitor" "foo" {
...
-		i, _ := strconv.Atoi(r.Primary.ID)
-		if _, err := client.GetMonitor(i); err != nil {
-			if strings.Contains(err.Error(), "404 Not Found") {
+		i, _ := strconv.ParseInt(r.Primary.ID, 10, 64)
+		_, httpresp, err := datadogClientV1.MonitorsApi.GetMonitor(authV1, i).Execute()
+		if err != nil {
+			if httpresp != nil && httpresp.StatusCode == 404 {
 				continue
 			}
 			return fmt.Errorf("received an error retrieving monitor %s", err)
```
Comment (zippolyte) [inline datadog/resource_datadog_monitor_test.go:1269]:
```suggestion
			return translateClientError(err, "error retrieving monitor")
```

File: datadog/resource_datadog_monitor_test.go [hunk 14]
```diff
@@ -1254,24 +1257,28 @@ resource "datadog_monitor" "foo" {
...
 
-func existsHelper(s *terraform.State, client *datadog.Client) error {
+func existsHelper(s *terraform.State, datadogClientV1 *datadogV1.APIClient, authV1 context.Context) error {
 	for _, r := range s.RootModule().Resources {
-		i, _ := strconv.Atoi(r.Primary.ID)
-		if _, err := client.GetMonitor(i); err != nil {
+		i, _ := strconv.ParseInt(r.Primary.ID, 10, 64)
+		_, _, err := datadogClientV1.MonitorsApi.GetMonitor(authV1, i).Execute()
+		if err != nil {
 			return fmt.Errorf("received an error retrieving monitor %s", err)
```
Comment (zippolyte) [inline datadog/resource_datadog_monitor_test.go:1282]:
```suggestion
			return translateClientError(err, "error retrieving monitor")
```

**Review by zippolyte — CHANGES_REQUESTED** [review]
Just a few small changes to have better error message when using the go client

**Review by zippolyte — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**